control-plane/plugin/src/resources/node.rs

```diff
@@ -105,18 +111,98 @@ impl List for Nodes {
     }
 }
 
+/// The NodeDisplayLabels structure is responsible for controlling the display formatting of Node
+/// objects. `#[serde(flatten)]` and `#[serde(skip)]` attributes are used to ensure that when the
+/// object is serialised, only the `inner` object is represented.
+#[derive(Serialize, Debug)]
+pub struct NodeDisplayLabels {
+    #[serde(flatten)]
+    pub inner: Vec<openapi::models::Node>,
+    #[serde(skip)]
+    show_labels: bool,
+}
+
+impl NodeDisplayLabels {
+    /// Create a new `NodeDisplayLabels` instance.
+    pub(crate) fn new(node: openapi::models::Node, show_labels: bool) -> Self {
+        let vec: Vec<openapi::models::Node> = vec![node];
+        Self {
+            inner: vec,
+            show_labels,
+        }
+    }
+    /// Get a list of node labels.
+    pub(crate) fn get_node_label_list(node: &openapi::models::Node) -> Vec<String> {
+        let mut node_labels: Vec<String> = vec![];
+
+        match &node.spec {
+            Some(ns) => match &ns.labels {
+                Some(ds) => {
+                    node_labels = ds
+                        .iter()
+                        .map(|(key, value)| format!("{}={}", key, value))
+                        .collect();
+                }
+                None => {}
+            },
+            None => {}
+        }
+        node_labels
+    }
+}
+
+// Create the header for a `NodeDisplayLabels` object.
+impl GetHeaderRow for NodeDisplayLabels {
+    fn get_header_row(&self) -> Row {
+        let mut header = (*utils::NODE_HEADERS).clone();
+        if self.show_labels {
+            header.extend(vec!["LABELS"]);
+        }
+        header
+    }
+}
+
+impl CreateRows for NodeDisplayLabels {
+    fn create_rows(&self) -> Vec<Row> {
+        let mut rows = vec![];
+        if self.show_labels {
+            for node in self.inner.iter() {
+                let mut row = node.create_rows();
+                let labelstring = NodeDisplayLabels::get_node_label_list(node).join(", ");
+                // Add the node labels to each row.
+                row[0].add_cell(Cell::new(&labelstring));
+                rows.push(row[0].clone());
+            }
+        }
+        rows
+    }
+}
+
 /// Node resource.
 #[derive(clap::Args, Debug)]
 pub struct Node {}
 
 #[async_trait(?Send)]
-impl Get for Node {
+impl GetWithArgs for Node {
     type ID = NodeId;
-    async fn get(id: &Self::ID, output: &utils::OutputFormat) -> PluginResult {
+    type Args = GetNodeArgs;
+    async fn get(id: &Self::ID, args: &Self::Args, output: &utils::OutputFormat) -> PluginResult {
         match RestClient::client().nodes_api().get_node(id).await {
             Ok(node) => {
-                // Print table, json or yaml based on output format.
-                utils::print_table(output, node.into_body());
+                let node_display =
```

Comment:
There's a mix here of using the NodeDisplayLabels and not using it.
IMHO we should either use it only when show_labels true or then always use it and don't need to check args again

Reply:
done